src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,143 @@
+        // now, if we already have some install data, we are going to be checking the variant we're about to install
+        // versus two different things:
+        // 1. is it in the expected list?
+        // 2. is it already NOT in the installed list?
+        // we'll just do nothing if it fails either of these and return false.
+        val retVal = if (currentInstallData != null) {
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\talready had install data !!!", "HullmodExoticHandler")
+            // now, check if this variant is in the expected list
+            val expectedVariants = currentInstallData.listOfExpectedVariants
+//            val isExpected: Boolean = expectedVariants.contains(variant)
+            // obviously these don't match
+            val isExpected: Boolean = expectedVariants
+                    .map { variant -> variant.hullSpec.hullId }
+                    .contains(variant.hullSpec.hullId)
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\tisExpected: ${isExpected}", "HullmodExoticHandler")
+
+            val variantsWeAlreadyInstalledOn = currentInstallData.listOfVariantsWeInstalledOn
+            val hasNotInstalledOnThisVariantAlready = variantsWeAlreadyInstalledOn.contains(variant).not()
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\thasNotInstalledOnThisAlready: ${hasNotInstalledOnThisVariantAlready}", "HullmodExoticHandler")
+
+            // Since I don't know how to check anything with parentFleetMember yet, lets just leave it out of the equation for now
+            isExpected && hasNotInstalledOnThisVariantAlready
+        } else {
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\tthere was no install data for this key ${hullmodExoticKey}", "HullmodExoticHandler")
+            // If we don't have any install data in the map, this is going to be simple. The optional variant list must be present
+            // If not present, just... do nothing for now, because we can't start off from a submodule I hope. So for now, throw
+            if (variantList.isPresent().not()) {
+                throw IllegalStateException("Since there is no HullmodExoticInstallData for these parameters, the Optional must be present!")
+            }
+
+            // And now, add stuff to the map ...
+            lookupMap[hullmodExoticKey] = HullmodExoticInstallData(
+                    parentFleetMemberAPI = parentFleetMember,
+                    listOfExpectedVariants =  variantList.get(),
+                    listOfVariantsWeInstalledOn = emptyList()
+            )
+
+            // And obviously return true
+            true
+        }
+
+        AnonymousLogger.log("<-- shouldInstallHullmodExoticToVariant()\treturning ${retVal}", "HullmodExoticHandler")
+        return retVal
+    }
+
+
+    fun installHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI): Boolean {
```

Comment:
javadoc.

also, two remarks:
1) why is this "installing" it just into the `lookupMap` and not also performing the actual installation as done by the `HullmodExotic` ?
2) can the number of "steps" be reduced so that we just shoot one fire-and-forget method and it doing everything?

Reply:
ok, so things have improved. a little bit.

the `removeHullmodExoticFromFleetMember(...)` is a step in this direction. 
but whether the `installHullmodExoticToFleetMember(...)` should replace the majority of HullmodExotic::onInstall() is a different question I don't have a good call on yet.

perhaps it could. but should it? only one way to find out - lets try it out and see.

Reply:
no, this whole idea was actually done with the "flows" and the callbacks in the `HullmodExotic` so... no, we will not further streamline it into the `HullmodExoticHandler`. It's reason for existing is clearly outlined and it's not going to meddle into all other classes (`HullmodExotic` and `ExoticHullmod`) to do their work or call their methods on them.

The `HullmodExotic` remains the key player. It invokes these "flows" and then maybe calls the `ExoticHullmod`'s unapply() method where necessary.

Reply:
resolving.